### ISSUE TITLE
[FEATURE] Better question submission flow

### ISSUE THREAD

**Issue by ykdojo** [ykdojo/defaang#216]
### Description

When a user submits a question, there should be a better indication to show that the question has been submitted successfully.

**Comment by epicadidash** [comment]
@ykdojo would love to work on this

**Comment by adilwahla** [comment]
there should be some kind of confirmation that yes your questions has been submitted successfully thanks  @ykdojo for early detection of issue :) i just submitted question and found this issue ...looking to contribute in other things

**Comment by ykdojo** [comment]
@adilwahla thank you so much. Would you like to take this issue though?

**Comment by adilwahla** [comment]
Sure if you give me bit direction as well 🤠

**Comment by ykdojo** [comment]
@adilwahla how would you approach it if it were up to you?

**Comment by adilwahla** [comment]
@ykdojo i would make button size increased horizontally and upon successful question submission it will turn green else red your thoughts please! 
open to suggestions as well

**Comment by adilwahla** [comment]
used submit message ...can make div pretty 
![Screenshot (23)](https://user-images.githubusercontent.com/48839911/193684233-b1e3ce16-de92-40be-bb35-fe9d6ddb0447.png)

**Comment by epicadidash** [comment]
> used submit message ...can make div pretty ![Screenshot (23)](https://user-images.githubusercontent.com/48839911/193684233-b1e3ce16-de92-40be-bb35-fe9d6ddb0447.png)

It will not look good , We can show it at the top instead

**Comment by ykdojo** [comment]
@adilwahla what do you think of transitioning to a different page entirely?

**Comment by adilwahla** [comment]
ok new page should show message like "Successfully submitted" would you like to add more questions? ->button to come back 
does it makes sense?

**Comment by epicadidash** [comment]
Yes

**Comment by adilwahla** [comment]
@ykdojo  its done ! please have a look I am opening PR 

![Screenshot (25)](https://user-images.githubusercontent.com/48839911/194082030-0ea5ae1d-49d4-4be0-b1c2-d76aaeb06fa4.png)

**Comment by adilwahla** [comment]
please check out i have made video 
https://www.awesomescreenshot.com/video/11525877?key=371821889405da656419b58003392b21

**Comment by adilwahla** [comment]
> @adilwahla what do you think of transitioning to a different page entirely?

followed this & i am just thinking we can add this transition to when every page to page transition on dynamic route! does it makes sense

**Comment by ykdojo** [comment]
Checked the video. After the transition, I think it should go to a separate page to indicate that the question has been submitted successfully?

**Comment by Akshay1018** [comment]
My suggestion is to add a popup for the successful confirmation and we can add the button(Add more?) on the popup So that users can add more questions. If the user clicks on Add more we will redirect to the /questions/new page.

**Comment by Akshay1018** [comment]
@ykdojo My approach is to check the status of the response that we are getting after making an API call on submitting the form. If the response status is 200 then we can show the Modal(i.e. popup) otherwise errors will be displayed.

**Comment by adilwahla** [comment]
> Checked the video. After the transition, I think it should go to a separate page to indicate that the question has been submitted successfully?

sure I will  update it

**Comment by ykdojo** [comment]
@Akshay1018 that approach might work, too. @adilwahla, what do you think?

**Comment by adilwahla** [comment]
yes sure

**Comment by Akshay1018** [comment]
@ykdojo shall I take this issue and work on it?

**Comment by ykdojo** [comment]
@Akshay1018 please decide with @adilwahla, but yes feel free to start working on it

**Comment by adilwahla** [comment]
![Screenshot (26)](https://user-images.githubusercontent.com/48839911/194399576-693338e6-08f2-460b-85ec-bd346930dc03.png)

**Comment by adilwahla** [comment]
@Akshay1018 together we will add value to DEFAANG I love pair Programming <3

**Comment by adilwahla** [comment]
@ykdojo checkout please 
https://www.awesomescreenshot.com/video/11558985?key=f6b3c2d268ed28298fb820b5c196deb5

**Comment by Akshay1018** [comment]
@adilwahla I looked at the video and the message is great but I'm a bit curious to know do we need to show the success message on a different route(/success). why not be on the same page?

**Comment by adilwahla** [comment]
@adilwahla the reason is we want to make question submit flow better &  that the user should feel like he has achieved something what he intended for @ykdojo  has similar suggestions !
Now one point is raised should we add another button for back to home if user doesn't chose to submit another question @ykdojo  your thought? plz

**Comment by Akshay1018** [comment]
![defaang](https://user-images.githubusercontent.com/62614172/194411187-8f1b72bc-7fae-473c-9c52-a087008b32f9.png)

**Comment by Akshay1018** [comment]
@adilwahla this is how the page will look like. when you will click on submit, it will check the response code, and based on the success code it will show a message like this. By clicking on Add more? button it will land us on the same page.

If we are adding back button on the /success page then again we will be on the same page. I think we need to check whether we need an extra route or not, We can optimize by showing messages on the same route.

**Comment by Akshay1018** [comment]
I will style this popup by adding a success icon and button also.

**Comment by adilwahla** [comment]
I did that previously but I think moving to new page is better idea i guess; it gives a better flow; just think for layman submitting question. Secondly Defaang is going to make further theme changes & i think new page suits 
@ykdojo  can give better idea !

**Comment by Akshay1018** [comment]
@adilwahla Yes, I think two options we have now. @ykdojo any luck with this solution?

**Comment by ykdojo** [comment]
I think this one looks pretty clear:
![image](https://user-images.githubusercontent.com/1811651/194605075-b6dc6c3a-e429-4736-a9d4-9db86c5931a8.png)

**Comment by Akshay1018** [comment]
@ykdojo Thanks, I will test, commit the code, and will make PR soon. You can test on your end and merge.

**Comment by adilwahla** [comment]
@ykdojo  what if user doesn't want to further submit question & want to move on discussion page or home page ?
i think we should let users to navigate other pages as well your thoughts ?

**Comment by Akshay1018** [comment]
@adilwahla if the user wants to add more questions, the user can click on "Add more?" button and will be rendered on the same page. Coming to the discussion page and home page, I think we need to add the navbar on every page, by clicking on the product logo, the user will be navigated to the home page. For the discussion section, we can add a separate card on the right side of the main section. @ykdojo What do you think?

**Comment by ykdojo** [comment]
@Akshay1018 sounds good! It's hard to tell without having a visual element though.

**Comment by Akshay1018** [comment]
@ykdojo I'll work on it and update the progress.

**Comment by Akshay1018** [comment]
@ykdojo by the way, do we need a discussion section? if yes, then where it should be?

**Comment by adilwahla** [comment]
@ykdojo actually we need to add menu items in navigation bar >problems then submenu will have description>submission>Discusss>solution 
should i make figma first?
